### DISCUSSION TITLE
RFC: Dashboard API Docs 2.0

### DISCUSSION THREAD

**Discussion by saltcod** [supabase/supabase#18038] [General]
The goal of this RFC (Request for Comment) is to identify how developers are currently using the API documentation page in the Supabase Studio (https://supabase.com/dashboard/project/_/api). We’re working on a revision of these docs, we want to make sure we have a better sense of how developers are using them.

### Motivation
We want to make it as fast as possible to get started with new projects. At the same time, we want to make it as easy as possible to grab the code you need to start querying your database. The current api docs don’t achieve either of these goals.

### Where we are 
We’ve just released new revision of the API docs, making them globally available from anywhere in the dashboard as a side panel. This is opt-in at the moment. You can enable it from the new Feature Previews modal:

<p float="left">
<img align="top" width="300" src="https://github.com/supabase/supabase/assets/105593/e143bab3-9594-431e-9246-d2612155d2bf" />
<img align="top" width="550" alt="screenshot-2023-10-09-at-11 00 25" src="https://github.com/supabase/supabase/assets/105593/aa91d741-fb2e-4b1d-bfa7-43a995897689">
</p>

This is a small, incremental improvement over what we had. We didn't move much around, and planned on this iteration to be relatively small. We want to gather feedback from users before we really dive in here with a proper 2.0 iteration. 

### What we currently know
- Our project connection settings and api docs are in multiple places
- Our api docs page isn’t heavily used
- Users like having the API Preview panel in the Table Editor for easy access

### What we’d like from you
We’d love to get your feedback on:

- How you use the API Docs in your work?
- Do you use the API Docs page?
- Do you use the API Preview side panel in the Table Editor?
- Do you use neither of these and use the official docs site instead?
- What would you like to see in this panel that isn’t in the docs?
- Are there specific parts of the reference docs that would be useful in the side panel?

**Comment by aiherrera** [comment]
Oh, you guys did it again 🚀

- How you use the API Docs in your work?
Usually when I'm coding a quick prototype I go directly to the corresponding section of the API for the code example, with the time I also go less than at the beginning, but that's obvious, that will vary depending on how much a developer had interacted with the Supabase service.

- Do you use the API Preview side panel in the Table Editor?
Just enabled and love it!

- Do you use neither of these and use the official docs site instead?
I use both, but honestly I spend more time in the official docs

- What would you like to see in this panel that isn’t in the docs?
- Are there specific parts of the reference docs that would be useful in the side panel?
Not sure about these two, if I came up with something I'll update my comment.

![image](https://github.com/supabase/supabase/assets/12240579/e38f26de-fa48-4bc6-a62c-3793beacd5dc)

**Reply by saltcod** [reply]
Thanks @aiherrera!  Appreciated!

**Comment by s0h311** [comment]
- How you use the API Docs in your work?
To look up the API of a function or which function I need in order to get the result I want.
- Do you use the API Docs page?
I have only used the https://supabase.com/docs page till now. I've using supabase for 8-9 months.
- Do you use the API Preview side panel in the Table Editor?
No, but now that I know it exists, I will use it more often
- Do you use neither of these and use the official docs site instead?
Until now, yes
-  What would you like to see in this panel that isn’t in the docs?
--
-  Are there specific parts of the reference docs that would be useful in the side panel?
--

**Reply by saltcod** [reply]
Thank you @s0h311! Appreciated!

**Comment by OmarAfet** [comment]
it would be cooler if could just copy the types instead of downloading the TS file here:
<img width="715" alt="image" src="https://github.com/user-attachments/assets/7106d91f-20b3-4be8-94c1-92a9212887dd">

**Comment by InfinityXone** [comment]
v0 is not.being generated
